src/Action/LoginAction.php

```diff
@@ -77,7 +83,8 @@ public function __construct(
         Pool $adminPool,
         TemplateRegistryInterface $templateRegistry,
         TokenStorageInterface $tokenStorage,
-        Session $session
+        Session $session,
+        TranslatorInterface $translator
```

Comment:
```suggestion
        ?TranslatorInterface $translator = null
```
Otherwise is BC-break, the class is final, but someone might be instantiating the class by their own.

Reply:
I see. I'm not yet used to thinking about things like that. :)